distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerProcess.java

```diff
@@ -230,6 +233,18 @@ private static Process createProcess(
         return processStarter.start(builder);
     }
 
+    /** Tells whether to disable the incubating vector module - it's enabled by default on JDK 20. */
+    static boolean disableIncubatorModule(List<String> jvmOptions) {
+        if (Runtime.version().feature() != 20) {
+            // we currently only support JDK 20. Each JDK version should be enabled explicitly, as appropriate
+            return true;
+        }
+        if (jvmOptions.contains("-Des.disable.jdk.incubator.vector")) {
```

Comment:
This is not a "real" system property, just an indication to the CLI launcher. Maybe there is a better way to pass this to the launcher @rjernst ?

Reply:
I wonder if we really need a configuration option here at all? Instead of hardcoding adding the incubating module here, could it be added to the default jvm.options? There is already java version guarding there, and then if a user wants to disable it, they remove it from their options file. Note this isn't perfect because of packaging complexities where we advise against modifying jvm.options directly, and in this case there is not way to disable once it added to the command line. However, I think jvm.options are a better place in general because that is how we allow users to modify jvm options, not with settings or system properties.

Reply:
Good idea. Done. eae28f4